components/task-details.tsx

```diff
@@ -1,9 +1,47 @@
 'use client'
 
 import { useTasks } from "@/context/task-context";
+import { useState } from 'react';
+import ModeEdit from "@mui/icons-material/ModeEdit";
+import Check from "@mui/icons-material/Check";
+
+interface Task {
+    id: string;
+    name: string;
+    description: string;
+    status: string;
+    createdAt: Date;
+}
 
 export const TaskDetails = () => {
-  const { tasks } = useTasks();
+  const { tasks, updateTask } = useTasks();
+  const [editingTaskId, setEditingTaskId] = useState('');
+  const [editedTaskName, setEditedTaskName] = useState('');
```

Comment:
you could put all editedX states into one editedTask state object which you can update, improving readability